To: vim_dev@googlegroups.com
Subject: Patch 7.3.168
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.168
Problem:    When the second argument of input() contains a CR the text up to
	    that is used without asking the user. (Yasuhiro Matsumoto)
Solution:   Change CR, NL and ESC in the text to a space.
Files:	    src/getchar.c


*** ../vim-7.3.167/src/getchar.c	2011-03-22 13:07:19.000000000 +0100
--- src/getchar.c	2011-04-28 14:50:26.000000000 +0200
***************
*** 635,645 ****
--- 635,648 ----
  /*
   * Stuff "s" into the stuff buffer, leaving special key codes unmodified and
   * escaping other K_SPECIAL and CSI bytes.
+  * Change CR, LF and ESC into a space.
   */
      void
  stuffReadbuffSpec(s)
      char_u	*s;
  {
+     int c;
+ 
      while (*s != NUL)
      {
  	if (*s == K_SPECIAL && s[1] != NUL && s[2] != NUL)
***************
*** 649,659 ****
  	    s += 3;
  	}
  	else
  #ifdef FEAT_MBYTE
! 	    stuffcharReadbuff(mb_ptr2char_adv(&s));
  #else
! 	    stuffcharReadbuff(*s++);
  #endif
      }
  }
  #endif
--- 652,667 ----
  	    s += 3;
  	}
  	else
+ 	{
  #ifdef FEAT_MBYTE
! 	    c = mb_ptr2char_adv(&s);
  #else
! 	    c = *s++;
  #endif
+ 	    if (c == CAR || c == NL || c == ESC)
+ 		c = ' ';
+ 	    stuffcharReadbuff(c);
+ 	}
      }
  }
  #endif
*** ../vim-7.3.167/src/version.c	2011-04-28 17:27:05.000000000 +0200
--- src/version.c	2011-04-28 17:28:36.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     168,
  /**/

-- 
Everyone has a photographic memory. Some don't have film.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
